dataactcore/utils/jobQueue.py

```diff
@@ -22,6 +30,32 @@ def enqueue(jobID):
     return response.json()
 
 
+def generate_f_file(submission_id, job_id, interface_holder_class,
+                    timestamped_name, is_local):
+    """Write rows from fileF.generateFRows to an appropriate CSV. Here the
+    third parameter, interface_holder_class, is a bit of a hack. Importing
+    InterfaceHolder directly causes cyclic dependency woes, so we're passing
+    in a class"""
+    with job_app.app_context():
+        job_manager = interface_holder_class().jobDb
+
+        try:
+            rows_of_dicts = fileF.generateFRows(job_manager.session,
+                                                submission_id)
+            header = [key for key in fileF.mappings]    # keep order
+            body = []
+            for row in rows_of_dicts:
+                body.append([row[key] for key in header])
+
+            write_csv(timestamped_name, is_local, header, body)
+            job_manager.markJobStatus(job_id, "finished")
```

Comment:
Also, it appears that the usage of the interface holder above isn't consistent with the recent database changes, it may be a good idea to merge dev into this branch to make sure things still work before completing the PR?